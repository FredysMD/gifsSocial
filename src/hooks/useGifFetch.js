import { useEffect, useState } from 'react';
import  { getGifs } from '../helpers/getGif';


export const useGifFetch = ( category ) => {

    const [state, setstate] = useState({
        data: [],
        loading: true
    });

    useEffect( () => {
        getGifs( category ).then( imgs => {
            setstate({
                data: imgs,
                loading: false
            })  
        })
    }, [category]);

    return state;
}