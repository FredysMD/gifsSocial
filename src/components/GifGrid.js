import React from 'react';
import Prototype from 'prop-types';

import { useGifFetch } from '../hooks/useGifFetch';

import { GifItem } from './GifItem';


const GifGrid = ( { category }) => {

    const { data:images, loading } = useGifFetch( category );

    return (
        <>
          <h2>{ category }</h2>
          { loading && <p className = "animate__animated animate__fadeIn">Loading...</p> }
          <ol>
            {
                images.map( img => (
                    <GifItem 
                        key = {img.id}
                        { ...img }
                    />
                ))
            }
          </ol>

        </>
    )
}

GifGrid.prototype = {
    category: Prototype.string.isRequired
}

export default GifGrid;

