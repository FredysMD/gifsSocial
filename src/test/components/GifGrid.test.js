import React from 'react';
import { shallow } from "enzyme";

import GifGrid from '../../components/GifGrid';
import { useGifFetch } from '../../hooks/useGifFetch';

jest.mock('../../hooks/useGifFetch');

describe('Test to GifGrid component', () => {
    
    const category = 'Messi';


    test('should have GifGrid component', () => {
        

        useGifFetch.mockReturnValue( {
            data: [],
            loading: true
        });

        const wrapper = shallow( <GifGrid category = { category } /> ); 
        expect( wrapper ).toMatchSnapshot();

    });

    test('should show item when loading images ', () => {
       
        const gifs = [{
            id: 'Fredys',
            url:'https://fredys.com',
            title: 'Fredys CEO'
        }];

        useGifFetch.mockReturnValue( {
            data: gifs,
            loading: false
        });

        const wrapper = shallow( <GifGrid category = { category } /> ); 

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find( 'p' ).exists() ).toBe(false);
        expect( wrapper.find( 'GifItem' ).length ).toBe( gifs.length );

    });
    
});
