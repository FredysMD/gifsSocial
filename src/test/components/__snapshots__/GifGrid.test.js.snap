// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test to GifGrid component should have GifGrid component 1`] = `
<Fragment>
  <h2>
    Messi
  </h2>
  <p
    className="animate__animated animate__fadeIn"
  >
    Loading...
  </p>
  <ol />
</Fragment>
`;

exports[`Test to GifGrid component should show item when loading images  1`] = `
<Fragment>
  <h2>
    Messi
  </h2>
  <ol>
    <GifItem
      id="Fredys"
      key="Fredys"
      title="Fredys CEO"
      url="https://fredys.com"
    />
  </ol>
</Fragment>
`;
